apps/wing/src/commands/test/test.ts

```diff
@@ -255,10 +255,18 @@ async function testAwsCdk(synthDir: string, options: TestOptions): Promise<std.T
     await withSpinner("cdk deploy", () => awsCdkDeploy(synthDir));
 
     const [testRunner, tests] = await withSpinner("Setting up test runner...", async () => {
+      let stackName = process.env.CDK_STACK_NAME!;
+      // get the file name to concat with stack name
+      const match = synthDir.match(/\/(\w+)\./);
+      if (match) {
+        // remove special characters from file name
+        stackName += "-" + match[1].replace(/[^a-zA-Z0-9]/g, "");
+      }
+
       const testArns = await awsCdkOutput(
         synthDir,
         ENV_WING_TEST_RUNNER_FUNCTION_IDENTIFIERS_AWSCDK,
-        process.env.CDK_STACK_NAME!
+        stackName
```

Comment:
what if we have two files of the same name? (in different folders) would it be better to randomize it as we do to the rootId? (we can also use only valid letters in the randomization, keeping it simpler :) )

Reply:
Tsuf, the problem is that I cannot use a randomized value because I need to obtain this value in two different places, both in `wingsdk/target-awscdk/app` and in `wing/command/test`

A second problem is that if I structure it like this:

![image](https://github.com/winglang/wing/assets/67694075/a24c5c33-9f6a-4d65-bd36-8ccb421add5b)

The `env.test.w` is tested, but the tests in the `folder` do not run independently of the target (tf-aws, awscdk).

Reply:
Maybe randomize the stack-name on the test command, then pass it as an env var?
https://github.com/winglang/wing/blob/64d830eae5a4c8c2d6588c7c879c3fd88df6be9c/apps/wing/src/commands/test/test.ts#L99 (like we did to the rootId here)
https://github.com/winglang/wing/blob/54f03f7c0be88bf49b3e4e61bcb7196d3b04b9ed/libs/wingcompiler/src/compile.ts#L127-L129 (here it's added to the `process.env` of the context we run the preflight on)
and be put here as a prop: https://github.com/winglang/wing/blob/add9ab68be3aa49c13944a4148f5229e85ddf83f/libs/wingc/src/jsify.rs#L188

Reply:
yes, it can work :) 
we can also hash the absolute path to the wing entry file